straxen/plugins/events/event_waveform.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+import strax
+import straxen
+
+export, __all__ = strax.exporter()
+
+
+@export
+class EventWaveform(strax.LoopPlugin):
```

Comment:
Thanks Daniel, I will look into it!

Reply:
So in that case changing it into a regular plugin could make sense, although I am not sure. In case you are interested here is what you would do: 
1. replace the LoopPlugin by a regular `strax.Plugin` 
2.  replace `compute_loop` by the regular compute method. 
3. Inside that loop you would do something like
```python
selected_peaks = np.zeros(len(events), strax.time_fields)
selected_peaks['time'] = events['s1_time']
selected_peaks['endtime'] = events['s1_endtime']

fc_index = strax.fully_contained_in(peaks, selected_peaks)
_is_peak_in_event = fc_index != -1
fc_index  = fc_index [_is_peak_in_event ]
events[fc_index  ]['s1_data'] = peaks[_is_peak_in_event ]['data']
```

Reply:
Hi Daniel, thanks for the comments and I have updated both the `event_area_per_channel` and `event_waveform` plugins from `strax.LoopPlugin` to `strax.Plugin`. The performanced get improved by roughly 50%.